test/hack/resource/clean/main.go

```diff
@@ -88,7 +91,7 @@ func main() {
 				resourceLogger.Errorf("%v", err)
 			}
 			// Should only fire metrics if the resource have expired
-			if clusterName == "" {
+			if lo.FromPtr(clusterName) == "" {
```

Comment:
Shouldn't we fire a metric regardless of it was from a specific cluster? IIUC, this should fire if L89 doesn't fail, regardless of if it was cleaned up for a specific cluster or not.